torch/_dynamo/symbolic_convert.py

```diff
@@ -2649,14 +2649,14 @@ def YIELD_FROM(self, inst):
                 return
             try:
                 val = tos.next_variable(self)
+            except (StopIteration, exc.UserStopIteration):
+                self.pop()
+                return
+            else:
                 self.push(val)
                 # TODO(voz): Unclear if we need the push None in YIELD_VALUE?
                 self.YIELD_VALUE(inst)
                 self.pop()
-                self.push(tos)
```

Comment:
`tos` is not popped from the stack. So we should not push it into the stack again.